### ISSUE TITLE
Distinct on Composite User Type property fails

### ISSUE THREAD

**Issue by PleasantD** [nhibernate/nhibernate-core#2856]
This is a regression introduced in v5.3.0

If I define an `ICompositeUserType` that directly maps properties to columns, the LINQ query should act on the mapped type's columns.

Say I define `PhoneNumber` with `Number` and `Ext` properties and map them to separate columns.

Prior to v5.3.0 the following worked
```
using (var s = OpenSession())
{
    s.Save(new Entity { 
        Name = "Company",
        Phone = new PhoneNumber("745-555-1234"),
    });
    s.Save(new Entity
    {
        Name = "Bob",
        Phone = new PhoneNumber("745-555-1234", "x123"),
    });
    s.Save(new Entity
    {
        Name = "Jane",
        Phone = new PhoneNumber("745-555-1235"),
    });
    s.Flush();
}

using (var s = OpenSession())
{
    var numbers = s.Query<Entity>().Select(x => x.Phone.Number).Distinct().ToList();
    Assert.That(numbers.Count, Is.EqualTo(2));
}
```

After v5.3.0 the Count returned is 3.

Inspecting the SQL generated reveals that both columns of the composite type are being included in the generated SQL rather than the single selected column and thus the `DISTINCT` clause is returning unexpected results.

**Comment by PleasantD** [comment]
[GH2856.zip](https://github.com/nhibernate/nhibernate-core/files/6817530/GH2856.zip)
Here's a unit test to illustrate the above.

**Comment by bahusoid** [comment]
Does it work properly with hql? 
```C#
s.CreateQuery("select distinct x.Phone.Number from Entity x").List();
```

**Comment by PleasantD** [comment]
The HQL version works fine.
I updated (and fixed) the unit tests in my previous example.

**Comment by gliljas** [comment]
I can add that I've experienced something similar. When selecting specific columns (with a LINQ Select) all columns where included in the SQL when the mapping contained a custom user type.

**Comment by martin-the-geek** [comment]
One thing while researching a similar issue was that prior to 5.3.0 the SelectClauseNominator would identify the composite property expression as a candidate

https://github.com/nhibernate/nhibernate-core/blob/329d591b54fbc6faf9d4aa12d48ff522c2c86a4a/src/NHibernate/Linq/Visitors/SelectClauseNominator.cs#L172

After 5.3.0, it would only add the whole composite as a candidate

https://github.com/nhibernate/nhibernate-core/blob/a97627d8f1b06388aa68d4b0c5ddf386a0a1781b/src/NHibernate/Linq/Visitors/SelectClauseNominator.cs#L182-L185

**Comment by hazzik** [comment]
Fixed by #2877
